cpp/src/arrow/flight/sql/types.h

```diff
@@ -119,6 +119,14 @@ struct ARROW_FLIGHT_SQL_EXPORT SqlInfoOptions {
     /// If 0, there is no timeout.
     FLIGHT_SQL_SERVER_TRANSACTION_TIMEOUT = 101,
 
+    /// Retrieves a boolean value indicating whether the Flight SQL Server
+    /// supports executing bulk ingestion.
+    FLIGHT_SQL_SERVER_BULK_INGESTION = 102,
+
+    /// Retrieves a boolean value indicating whether the Flight SQL Server
+    /// supports transactions for bulk ingestion.
+    FLIGHT_SQL_SERVER_INGEST_TRANSACTIONS_SUPPORTED = 103,
```

Comment:
Why was there a jump from 9 to 100 above? Are there two categories? It would be nice if this were explained in the comments.

Reply:
I read through the [PR](https://github.com/apache/arrow/pull/13492) where the enum keys 9 and 100 were originally defined but couldn't find an explicit reason. The keys added in this PR seem to make sense in the 0-500 range but I'm not sure whether they should be added after 9 or after 101. Any context on this @lidavidm?

Reply:
There's not a real reason, it just looks a little cleaner in the case where we add other keys in the future.

Reply:
I don't understand what is "a lot cleaner" here? Why is it cleaner to add these options after 100 rather than after 9?

Reply:
Sorry, I just realized I was looking at a different part of the file. I don't recall why we jumped to 100 originally. But since 100 and 101 are timeouts we could make these new codes 10 and 11 instead of 102 and 103. Is that better?

Reply:
Yes, that would seem more logical to me.

Reply:
Makes sense, I changed these to 10 and 11.